### PR TITLE
Stop using 4bytes for contract deployment

### PR DESCRIPTION
**Pull request by FrederikBolding**
## Explanation
This PR fixes a bug where we would try to show called method names for contract deployments. Since contract deployments have the bytecode as the data we can't look up method signatures using the first 4 bytes as you can with a normal contract interaction.

## Manual Testing Steps

1. Go to the test dapp
2. Deploy a contract
3. See that the transaction title is "contract deployment"

## Pre-Merge Checklist

- [x] PR template is filled out
- [x] Manual testing complete & passed
- [x] **IF** this PR fixes a bug, a test that _would have_ caught the bug has been added
- [x] PR is linked to the appropriate GitHub issue
- [x] **IF** QA attention is required, "QA Board" label has been applied
- [x] PR has been added to the appropriate release Milestone

### CODE REVIEW

**Review by kumavis — DISMISSED** [review]

**Comment by jpuri** [comment]
Can you plz add some description to the PR and if possible test coverage.

File: ui/hooks/useTransactionDisplayData.js [hunk 1]
```diff
@@ -217,17 +217,20 @@ export function useTransactionDisplayData(transactionGroup) {
...
-  } else if (
-    type === TRANSACTION_TYPES.DEPLOY_CONTRACT ||
-    type === TRANSACTION_TYPES.CONTRACT_INTERACTION
-  ) {
+  } else if (type === TRANSACTION_TYPES.CONTRACT_INTERACTION) {
     category = TRANSACTION_GROUP_CATEGORIES.INTERACTION;
     const transactionTypeTitle = getTransactionTypeTitle(t, type);
     title =
       (methodData?.name && camelCaseToCapitalize(methodData.name)) ||
       transactionTypeTitle;
```
Comment (adonesky1) [inline ui/hooks/useTransactionDisplayData.js:225]: Do we ever add `methodData` to a transaction where the `to` address is `null`/`undefined`?

Comment (FrederikBolding) [reply ui/hooks/useTransactionDisplayData.js:225]: We shouldn't. We should only use it for valid method calls to a contract.

**Comment by FrederikBolding** [comment]
> Can you plz add some description to the PR and if possible test coverage.

Added a description. I would add test coverage, but I don't see any fixtures for it? 🤔

**Review by brad-decker — APPROVED** [review]
LGTM

File: ui/hooks/useTransactionDisplayData.js (hunk 1 above)
Comment (adonesky1) [reply ui/hooks/useTransactionDisplayData.js:225]: I feel like ideally we would guarantee that methodData won't be attached to a `transaction` where address is empty or zero address further upstream (maybe in the `TransactionsController`), but won't block on it here...

Comment (adonesky1) [reply ui/hooks/useTransactionDisplayData.js:225]: I feel like ideally we would guarantee that methodData won't be attached to a transaction where address is empty or zero address further upstream (maybe in the TransactionsController), but won't block on it here...

**Review by adonesky1 — APPROVED** [review]

File: ui/hooks/useTransactionDisplayData.js (hunk 1 above)
Comment (brad-decker) [reply ui/hooks/useTransactionDisplayData.js:225]: in what scenario would we have a transaction without a 'to' address?

Comment (adonesky1) [reply ui/hooks/useTransactionDisplayData.js:225]: I think when you deploy a contract no?

Comment (brad-decker) [reply ui/hooks/useTransactionDisplayData.js:225]: 🤦🏻 yeah. that makes sense...

Comment (brad-decker) [reply ui/hooks/useTransactionDisplayData.js:225]: We don't attach methoData anywhere, its fetched at that top of the useTransactionDisplayData hook.

Comment (FrederikBolding) [reply ui/hooks/useTransactionDisplayData.js:225]: Yeah, it's not affiliated with the transaction outside of this hook afaik @adonesky1

**Pull request merged**